src/avk.cpp

```diff
@@ -7448,6 +7448,12 @@ using namespace cpplinq;
 			.setPSubpasses(result.mSubpasses.data())
 			.setDependencyCount(static_cast<uint32_t>(result.mSubpassDependencies.size()))
 			.setPDependencies(result.mSubpassDependencies.data());
+
+		// Maybe alter the createInfo itself?!
+		if (aAlterCreateInfoBeforeCreation) {
+			aAlterCreateInfoBeforeCreation(createInfo);
+		}

```

Comment:
I think, I'd prefer to add `vk::RenderPassCreateInfo` to `renderpass_t` as a member, add `renderpass_t::create_info()` getters and have `aAlterConfigBeforeCreation` be invoked at this point Only one `aAlterConfig*` would be required in that case. Is there anything that would speak against this approach?

Reply:
Adding a member and getter is fine; but I still think we need two `aAlter*` functions:
The original `aAlterConfigBeforeCreation` allows to modify stuff (e.g. attachment descriptions) that is subsequently used to **generate** the initial `createInfo`.
The new `aAlterCreateInfoBeforeCreation` then allows some final modifications to this structure. I don't see how these two can be combined into a single function without breaking existing code.